src/consts.rs

```diff
@@ -92,10 +84,8 @@ impl PartialEq for Constant {
         match (self, other) {
             (&Constant::Str(ref ls, ref lsty), &Constant::Str(ref rs, ref rsty)) => ls == rs && lsty == rsty,
             (&Constant::Binary(ref l), &Constant::Binary(ref r)) => l == r,
-            (&Constant::Byte(l), &Constant::Byte(r)) => l == r,
             (&Constant::Char(l), &Constant::Char(r)) => l == r,
-            (&Constant::Int(0, _, _), &Constant::Int(0, _, _)) => true,
-            (&Constant::Int(lv, _, lneg), &Constant::Int(rv, _, rneg)) => lv == rv && lneg == rneg,
+            (&Constant::Int(l), &Constant::Int(r)) => l == r,
```

Comment:
well... I can change that if we pass a `tcx`. But `Infer` infers to the other type before running any operation